### PR TITLE
sbojanic/ci improve security

### PR DESCRIPTION
**Pull request by sale3**
## Description (e.g. "Related to ...", etc.)
ci: replace tags with commit hashes

## Code review checklist (for code reviewer to complete)

- [ ] Pull request represents a single change (i.e. not fixing disparate/unrelated things in a single PR)
- [ ] Title summarizes what is changing
- [ ] Commit messages are meaningful (see [this][commit messages] for details)
- [ ] Tests have been included and/or updated, as appropriate
- [ ] Docstrings have been included and/or updated, as appropriate
- [ ] Changelog has been updated, as needed (see [CHANGELOG.md][changelog])

[changelog]: https://github.com/openlawlibrary/taf/blob/master/CHANGELOG.md
[commit messages]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

**Comment by renatav** [comment]
@sale3  Could you please provide some additional context, why is it better to hardcode the commit hashes?

**Review by n-dusan — COMMENTED** [review]
Seems reasonable. We lose some of legibility of tag information that we had. But not really sure how useful that was. @renatav could you make the final call?

**Comment by renatav** [comment]
It looks like this is a rule of thumb:
- If stability and security are your highest priorities (e.g., in critical production workflows), use a specific commit hash.
- If you prefer easier maintenance and can accept the minimal risk of changes introduced by updates, use a version tag.

I can say that I was not keeping up with the updates and only updated these major versions while working on the deployment of wheels. I'd say that we are neither gaining nor loosing that much, so we can merge this in. How do we set a reminder to keep track of the updates and replace that hash with a newer one fairly regularly?

**Comment by n-dusan** [comment]
We could potentially add a comment next to the commit hash indicating from which tag (e.g. v4) it got taken off of?

File: .github/workflows/ci.yml
Comment (n-dusan) [inline .github/workflows/ci.yml:35]:
```suggestion
      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 #@v4
```

**Review by n-dusan — COMMENTED** [review]
Thoughts? @renatav

**Comment by renatav** [comment]
@sale3  Could you address the comment that @n-dusan left, so that we can get this merged in and released before we merge the big TUF update PR

**Review by renatav — APPROVED** [review]

**Pull request merged**
